src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs

```diff
@@ -158,6 +160,101 @@ internal static TFSession LoadTFSession(IExceptionContext ectx, byte[] modelByte
             return new TFSession(graph);
         }
 
+        private static TFSession LoadTFSession(IHostEnvironment env, string exportDirSavedModel)
+        {
+            env.CheckValue(exportDirSavedModel, nameof(exportDirSavedModel));
+            var sessionOptions = new TFSessionOptions();
+            var exportDir = exportDirSavedModel;
+            var tags = new string[] { "serve" };
+            var graph = new TFGraph();
+            var metaGraphDef = new TFBuffer();
+
+            return TFSession.FromSavedModel(sessionOptions, null, exportDir, tags, graph, metaGraphDef);
+        }
+        // A TensorFlow frozen model is a single file. An un-frozen (SavedModel) on the other hand has a well-defined folder structure.
+        // Given a modelPath, this utility method determines if we should treat it as a SavedModel or not
+        internal static bool IsSavedModel(string modelPath)
+        {
+            FileAttributes attr = File.GetAttributes(modelPath);
+            return attr.HasFlag(FileAttributes.Directory);
+        }
+
+        internal static void CreateTempDirectory(string tempDirPath)
```

Comment:
>CreateTempDirectory [](start = 29, length = 19)

What's the main reason for this? Why we treat Windows differently?
If this is necessary, why it's part of tensorflow only and not part of utility which we use among whole project?

Reply:
(replied on other comment about its necessity)

currently used in TfTransform only, but yes I think bettwe place would be to make it part of utility for whole project. 

---
In reply to: [219577130](https://github.com/dotnet/machinelearning/pull/853#discussion_r219577130) [](ancestors = 219577130)

Reply:
Can you put comment with description why we doing this?
I would be really tempted to remove this code if I wouldn't know story behind it.

---
In reply to: [219578542](https://github.com/dotnet/machinelearning/pull/853#discussion_r219578542) [](ancestors = 219578542,219577130)